hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenAirdropsHandler.java

```diff
@@ -71,7 +87,104 @@ public void handle(@NonNull final HandleContext context) throws HandleException
     @NonNull
     @Override
     public Fees calculateFees(@NonNull final FeeContext feeContext) {
-        // todo calculate fees
-        return Fees.FREE;
+        final var body = feeContext.body();
+        final var op = body.tokenAirdrop();
+
+        final var defaultAirdropFees = feeContext.feeCalculator(SubType.DEFAULT).calculate();
+        // TODO: add a comment why do we need that
+        final var cryptoTransferFees =
+                CryptoTransferHandler.calculateCryptoTransferFees(feeContext, null, op.tokenTransfers());
```

Comment:
We can call `dispatchComputeFees` here also, to follow same pattern for associations also

Reply:
Done 👍